app/services/PenaltiesFrontendService.scala

```diff
@@ -16,29 +16,28 @@
 
 package services
 
-import models.getFinancialDetails.GetFinancialDetails
+import models.getFinancialDetails.{FinancialDetails, MainTransactionEnum}
 import models.getPenaltyDetails.GetPenaltyDetails
 import models.getPenaltyDetails.latePayment.{LPPDetailsMetadata, LPPPenaltyCategoryEnum, LatePaymentPenalty}
-import models.mainTransaction.MainTransactionEnum
 
 import javax.inject.Inject
 
 class PenaltiesFrontendService @Inject()() {
-  def combineAPIData(penaltyDetails: GetPenaltyDetails, financialDetails: GetFinancialDetails): GetPenaltyDetails = {
+  def combineAPIData(penaltyDetails: GetPenaltyDetails, financialDetails: FinancialDetails): GetPenaltyDetails = {
     val newLPPDetails = penaltyDetails.latePaymentPenalty.flatMap(
       _.details.map(_.map(
         oldLPPDetails => {
           val isAdditional = oldLPPDetails.penaltyCategory.equals(LPPPenaltyCategoryEnum.SecondPenalty)
-          val firstAndMaybeSecondPenalty = financialDetails.financialDetails.filter(_.chargeReference.exists(_.equals(oldLPPDetails.principalChargeReference)))
+          val firstAndMaybeSecondPenalty = financialDetails.documentDetails.get.filter(_.chargeReferenceNumber.exists(_.equals(oldLPPDetails.principalChargeReference)))
           val penaltyToCopy = firstAndMaybeSecondPenalty.find(lpp => {
-              if(isAdditional) MainTransactionEnum.secondCharges.contains(lpp.mainTransaction.get)
-              else MainTransactionEnum.firstCharges.contains(lpp.mainTransaction.get)
+              if(isAdditional) MainTransactionEnum.secondCharges.contains(lpp.lineItemDetails.get.head.mainTransaction.get)
+              else MainTransactionEnum.firstCharges.contains(lpp.lineItemDetails.get.head.mainTransaction.get)
```

Comment:
Adding this for traceability, we are going to ask ETMP to see if LPP1 and LPP2 come under different charges